extra_data/components.py

```diff
@@ -871,7 +881,12 @@ def buffer_shape(self, module_gaps=False, roi=()):
         interest on the inner dimensions of the data.
         """
         module_dim = self.det.n_modules if module_gaps else len(self.modno_to_keydata)
```

Comment:
While it is required here, it almost makes me wonder if it would be useful as standalone method `module_dim(module_gaps=False)`. It's indirectly possible noe with buffer_shape of course, or is there another way to figure out how many modules are there?

Reply:
`det.n_modules` tells you how many modules the detector should have. `det[key].shape[0]` is how many modules we have this data for. I'd rather not make another public way to do this, and I prefer a tiny bit of duplication to the extra complexity of another private method.